js/client-teambuilder.js

```diff
@@ -2759,85 +2786,12 @@
 			}
 			return false;
 		},
-		chooseMove: function (moveName, resetSpeed) {
+		chooseMove: function (moveName) {
 			var set = this.curSet;
-			if (!set) return;
-			var gen = this.curTeam.gen;
-
-			var minSpe;
-			if (resetSpeed) minSpe = false;
-			if (moveName.substr(0, 13) === 'Hidden Power ') {
-				if (!this.canHyperTrain(set)) {
-					var hpType = moveName.substr(13);
-
-					set.ivs = {hp: 31, atk: 31, def: 31, spa: 31, spd: 31, spe: 31};
-					if (this.curTeam.gen > 2) {
-						for (var i in exports.BattleTypeChart[hpType].HPivs) {
-							set.ivs[i] = exports.BattleTypeChart[hpType].HPivs[i];
-						}
-					} else {
-						for (var i in exports.BattleTypeChart[hpType].HPdvs) {
-							set.ivs[i] = exports.BattleTypeChart[hpType].HPdvs[i] * 2;
-						}
-						var atkDV = Math.floor(set.ivs.atk / 2);
-						var defDV = Math.floor(set.ivs.def / 2);
-						var speDV = Math.floor(set.ivs.spe / 2);
-						var spcDV = Math.floor(set.ivs.spa / 2);
-						var expectedHpDV = (atkDV % 2) * 8 + (defDV % 2) * 4 + (speDV % 2) * 2 + (spcDV % 2);
-						set.ivs.hp = expectedHpDV * 2;
-						if (set.ivs.hp === 30) set.ivs.hp = 31;
-					}
-				}
-			} else if (moveName === 'Return') {
-				this.curSet.happiness = 255;
+			if (moveName === 'Return') {
+				set.happiness = 255;
 			} else if (moveName === 'Frustration') {
-				this.curSet.happiness = 0;
-			} else if (moveName === 'Gyro Ball') {
-				minSpe = true;
-			}
-
-			if (this.curTeam.format === 'gen7hiddentype') return;
-
-			var minAtk = true;
-			if (set.ability === 'Battle Bond') minAtk = false; // only available through an event with 31 Atk IVs
-			var hpModulo = (this.curTeam.gen >= 6 ? 2 : 4);
```

Comment:
I didn't carry this part over to the new method `getAutoIVs`. Shouldn't it always be 2 since Hidden Power only depends on the parity of the IV?